hbase-testing-util/src/main/java/org/apache/hadoop/hbase/testing/TestingHBaseClusterImpl.java

```diff
@@ -172,4 +179,38 @@ public void startRegionServer() throws Exception {
   public void startRegionServer(String hostname, int port) throws Exception {
     util.getMiniHBaseCluster().startRegionServer(hostname, port);
   }
+
+  @Override
+  public Optional<ServerName> getActiveMasterAddress() {
+    return Optional.ofNullable(util.getMiniHBaseCluster().getMaster()).map(HMaster::getServerName);
+  }
+
+  @Override
+  public List<ServerName> getBackupMasterAddresses() {
+    return util.getMiniHBaseCluster().getMasterThreads().stream().map(MasterThread::getMaster)
+      .filter(m -> !m.isActiveMaster()).map(HMaster::getServerName).collect(Collectors.toList());
+  }
+
+  @Override
+  public List<ServerName> getRegionServerAddresses() {
+    return util.getMiniHBaseCluster().getRegionServerThreads().stream()
+      .map(t -> t.getRegionServer().getServerName()).collect(Collectors.toList());
+  }
+
+  @Override
+  public Optional<Region> getRegion(RegionInfo regionInfo) {
+    for (RegionServerThread t : util.getMiniHBaseCluster().getRegionServerThreads()) {
+      for (HRegion region : t.getRegionServer().getRegions()) {
+        if (region.getRegionInfo().equals(regionInfo)) {
+          return Optional.of(region);
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public Optional<OnlineRegions> getOnlineRegionsInterface(ServerName serverName) {
+    return Optional.ofNullable(util.getMiniHBaseCluster().getRegionServer(serverName));
```

Comment:
The HRegionServers extends RegionServerServices, RegionServerServices extends MutableOnlineRegions, and finally MutableOnlineRegions extends OnlineRegions, so HRegionServer itself is 'OnlineRegions', we do not need to go to RS coprocessor to get it. The design here is not give CP implementations a way to get the Region interface, but do not want to give the CP implementations the ability to add or remove from the online regions directly as it could mess up the state.